### ISSUE TITLE
Create data tables for plant nodes in `default` and `flowers`

### ISSUE THREAD

**Issue by Lowie375** [ubc-minetest-classroom/Minetest_Classroom#16]
## Status
See [the Plant Compendium project board](https://github.com/ubc-minetest-classroom/Minetest_Classroom/projects/2)

## Node Template
NOTES:
- All fields except `sci_name` are optional; do be warned, however, that leaving some blank or undefined may produce unexpected results
- `img_copyright` and `img_credit` are incompatible with one another (you can use one, but not both)
- `model_rot_horiz` and `model_rot_verti` take precedence over `model_rot_y` and `model_rot_x`
```lua
local template = {
    sci_name = "",          -- Scientific name of species
    com_name = "",          -- Common name of species
    fam_name = "",          -- Family name of species

    cons_status = {         -- Conservation statuses of species
        ns_global = "",         -- NatureServe global status
        ns_bc = "",             -- NatureServe BC status
        bc_list = ""            -- BC List (Red Blue List) status
    },
    tags = {""},            -- Tags representing various charactertistics of a species
    region = "",            -- Native region/range of species (displayed as "Found in [region]")
    height = "",            -- Species height
    bloom = "",             -- The way the species blooms
    more_info = "",         -- Extended description of species
    
    texture = {""},         -- Images of species (in `mod/textures`)
    model_obj = "",         -- Model file (in `mod/models`)
    model_rot_horiz = 0,    -- Initial horizontal rotation of model, about y-axis (in degrees; defaults to 180)
    model_rot_verti = 0,    -- Initial vertical rotation of model, about x-axis (in degrees; defaults to 0)
    model_rot_y = 0,        -- Alias for model_rot_horiz
    model_rot_x = 0,        -- Alias for model_rot_verti

    external_link = "",     -- Link to page with more species information
    img_copyright = "",     -- Copyright owner of species image (displayed as "Image (c) [img_copyright]")
    img_credit = "",        -- Author of species image (displayed as "Image courtesy of [img_credit]")
    info_source = "",       -- Source that species information was taken from
    last_updated = "",      -- Timestamp when species was last updated
}
-- Plant registration call
magnify.register_plant(template, {"mod:node", "mod:another_node", "other_mod:other_node"})
```

```lua
OBSOLETE PROPERTIES:
 -- model_spec = {}       -- Now pulled dynamically from model files, see "Notes on Model Use"
```

## Notes on Model Use
In order for MineTest to properly render models in formspecs, the following must be done to model files:
- Textures *must* be placed on group (`g`) lines
  - MineTest does not recognize `usemtl` lines when rendering models in formspecs
- Textures must appear exactly as their files are named

Example from [`magnify/magnify_default_nodes/magnify_default_pinus_contorta_var_latifolia.obj`](https://github.com/ubc-minetest-classroom/Minetest_Classroom/blob/magnify/mods/magnify/magnify_default_nodes/models/magnify_default_pinus_contorta_var_latifolia.obj#L1405-L1410):
```obj
...
f 220/721/5 218/714/5 217/713/5 219/722/5
g default_pine_tree_top.png
f 221/723/2 222/724/2 223/725/2 224/726/2
f 91/334/1 94/343/1 145/619/1 143/611/1
g default_pine_tree.png
f 223/727/3 222/724/3 225/728/3 226/729/3
...
```

**Comment by Lowie375** [comment]
Moved the status list to [https://github.com/ubc-minetest-classroom/Minetest_Classroom/projects/2](https://github.com/ubc-minetest-classroom/Minetest_Classroom/projects/2) so that anyone working on this can add and move things around as they are completed

**Comment by pauldpickell** [comment]
@Lowie375 is there a way to read the node types directly from the object? If this is possible, then it might  not be necessary for the user to specify the text list, since this list is just the node names.

**Comment by Lowie375** [comment]
I don't think node types are defined in node definition tables, but I could look and see if there's anything I could use from the plant generation code to automate plant registration

**Comment by Lowie375** [comment]
> @Lowie375 is there a way to read the node types directly from the object?

Just realized that you were referring to reading directly from the object files themselves - that could definitely be done with a file iterator, which shouldn't be too tough to implement. Will work on implementing that tomorrow
EDIT: This has been implemented as of commit [`7085390`](https://github.com/ubc-minetest-classroom/Minetest_Classroom/commit/70853901d6c70d0fa775a2a9049a146038a8ee0d)! Textures will now be pulled from the model files themselves rather than  the now-obsolete `model_spec` property

**Comment by Lowie375** [comment]
In the meantime, all models have been added to the data tables in `magnify_default_nodes` and `magnify_flowers_nodes` as of commit [`fb33a25`](https://github.com/ubc-minetest-classroom/Minetest_Classroom/commit/fb33a251fb7d27503b97be978089032f82b3a333) on `magnify`
